### DISCUSSION TITLE
Help us build a beautiful dashboard for Fonoster!

### DISCUSSION THREAD

**Discussion by psanders** [fonoster/yaps#646] [Announcements]
We're building a modern, beautiful dashboard to make building on Fonoster enjoyable. Please help us create a robust foundation that includes reusable components and intuitive interfaces for developers.  

![Component Library](https://github.com/user-attachments/assets/dedcecdd-6926-407a-a549-ed4300c224d9)

## Why Contribute?  

- Collaborate with a global community of developers and designers
- Gain recognition for your work in the open-source ecosystem
- Direct access to Fonoster's team
- Request a letter of recommendation from Fonoster's founder

## Current Tech Stack  

We're using modern tools to ensure a smooth and scalable development experience:  

- **React + TypeScript**: For building robust and type-safe components
- **Vite**: For fast and efficient development
- **MUI**: As the foundation of our design
- **Storybook**: For creating and testing reusable UI components
- **Fonoster's SDK**: To integrate with the admin APIs 

> We will provide contributors with access and API keys 

If you’re familiar with these technologies or eager to learn, we’d love to have you onboard!

## How to Get Started  

1. **Explore the designs**  
   - Review the [Core Components](https://www.figma.com/design/OsZlne0RvIgoFlFKF7hnAU/Shared-Component-Library?node-id=8-8505&node-type=canvas&t=xIEqMLiCbEFQGs33-0) in the Figma file
   - Check the [Dashboard designs](https://www.figma.com/design/8WQkuOgunwhDP791sHGso2/Fonoster-WebUI-2024?node-id=276-10356&node-type=canvas&t=sgsno2sV2jlRiuOs-0) in the Figma file
2. **Understand the workflow**  
   - Read the [CONTRIBUTING.md](https://github.com/fonoster/fonoster/blob/main/CONTRIBUTING.md) document
3. **Claim an issue**  
   - Find an unassigned issue labeled [dashboard](https://github.com/fonoster/fonoster/labels/dashboard)
   - Request to have it assigned to you
4. **Submit your work**  
   - Create a pull request (PR) for your contribution
   - Engage in feedback and iterate until it’s merged

Ready to contribute? We look forward to building something amazing with you!

**Comment by psanders** [comment]
### Getting started with Fonoster Storybook

To get started, clone the repository and run the following commands:

```bash
git clone https://github.com/fonoster/fonoster
```

Then, install the dependencies:

```bash
cd fonoster
npm install
npm run build
cp .env.example.dev .env
```

Finally, run the following command to start the Storybook server:

```bash
npm run start:storybook
```

Storybook will start a server and open a new tab in your browser.

**Comment by rolwin100** [comment]
Awesome 😎 we can start with installing material ui

**Reply by psanders** [reply]
@rolwin100, the early work I did with the webui already includes mui. 

The next step is creating GitHub issues, which I will work on tonight. 

However, it would be cool to code review the existing work in the `webui` module.

**Comment by psanders** [comment]
I just published the first few issues:

https://github.com/fonoster/fonoster/labels/dashboard

**Comment by obrucheoghene** [comment]
hi @psanders. I'm here to contribute

**Reply by psanders** [reply]
@obrucheoghene 

Now it is party 🙌

**Comment by psanders** [comment]
Quick update: I just added several GitHub issues covering components like Chip, GenericToggle, LinkBackTo, LinkBackPagination, OverviewCard, and more.

![image](https://github.com/user-attachments/assets/41c022b0-f39d-473f-a311-379f5595fa96)

https://github.com/fonoster/fonoster/issues?q=is%3Aissue+is%3Aopen+label%3Adashboard

**Comment by psanders** [comment]
Hey friends,

I create a track so everybody can see the progress of this workstream:

https://github.com/fonoster/fonoster/issues/675
